src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -310,22 +310,22 @@ static PrimaryKey peek(Object keys)
         {
             if (keys instanceof PrimaryKey)
                 return (PrimaryKey) keys;
-            if (keys instanceof SortedSetRangeIterator)
-                return ((SortedSetRangeIterator) keys).peek();
+            if (keys instanceof SortedSetKeyRangeIterator)
+                return ((SortedSetKeyRangeIterator) keys).peek();
 
             throw new AssertionError("Unreachable");
         }
     }
 
-    static class MergingRangeIterator extends RangeIterator
+    static class MergingKeyRangeIterator extends KeyRangeIterator
```

Comment:
@adelapena IntelliJ propagated the name change to few nested classes in few other classes like here. What do you think about this change? Is it reasonable to do (for me sounds reasonable)? What should be the name? I can see inconsistency in ASF, where in iterators each iterator class is prefixed by `KeyRange`, while in other files it's postfixed with `KeyRangeIterator`, see https://github.com/apache/cassandra/blob/7b33e91852aee7af2e995273237d0f3828cd9e4b/src/java/org/apache/cassandra/index/sai/memory/InMemoryKeyRangeIterator.java.

Reply:
I think it makes sense naming it `MergingKeyRangeIterator`. We can tolerate some inconsistency with ASF if it's for better inner consistency, IMO. I think this is still named in the already merged refactor for `main`, should we change it there too?

Reply:
I created #1414 to fix the same renaming in `main`. It also shows clearly these specific changes.